File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/config/model/ChainConfig.kt
package com.cinacoin.appkit.config.model

/**
 * Chain configuration data class
 */
data class ChainConfig(
    val chainId: Int,
    val name: String,
    val shortName: String,
    val symbol: String,
    val decimals: Int = 18,
    val rpcUrl: String,
    val explorerUrl: String,
    val iconUrl: String? = null,
    val testnet: Boolean = false
) {
    /**
     * Get CAIP-2 namespace string (e.g., "eip155:1")
     */
    fun toNamespaceString(): String = "eip155:$chainId"
}

File: packages/flutter-dart/lib/appkit_config/models/chain_config.dart
/// Chain configuration model
class ChainConfig {
  final int chainId;
  final String name;
  final String shortName;
  final String symbol;
  final int decimals;
  final String rpcUrl;
  final String explorerUrl;
  final String? iconUrl;
  final bool testnet;

  const ChainConfig({
    required this.chainId,
    required this.name,
    required this.shortName,
    required this.symbol,
    this.decimals = 18,
    required this.rpcUrl,
    required this.explorerUrl,
    this.iconUrl,
    this.testnet = false,
  });

  /// Get CAIP-2 namespace string (e.g., "eip155:1")
  String toNamespaceString() => 'eip155:$chainId';

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is ChainConfig &&
          runtimeType == other.runtimeType &&
          chainId == other.chainId;

  @override
  int get hashCode => chainId.hashCode;
}
